pkg/deploy/migrate-labels.go

```diff
@@ -0,0 +1,103 @@
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package deploy
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	cheFlavor = os.Getenv("CHE_FLAVOR")
+)
+
+// MigrateCheResourcesLabels searches for objects of different kinds in the cluster,
+// that are added by Che admin and used by the Operator in some way.
+// When such a resource is found, a new label should be added in order to have the resource cached.
+// This should scan all namespaces in order to support all namespaces mode.
+// If an error happens, warning is printed and execution continues.
+// Returns true if everything is updated without errors, false otherwise.
+func MigrateCheResourcesLabels(nonCachingClient client.Client) bool {
+	noErrors := true
+
+	// Prepare selector
+	chePartOfSelectorRequirement, err := labels.NewRequirement(KubernetesPartOfLabelKey, selection.Equals, []string{CheEclipseOrg})
+	if err != nil {
+		logrus.Error("Failed to create selector for resources migration. All custom user objects are lost.")
+		return false
+	}
+	partOfCheLebelSelector := labels.NewSelector().Add(*chePartOfSelectorRequirement)
+	listOptions := &client.ListOptions{
+		LabelSelector: partOfCheLebelSelector,
+	}
+
+	// Migrate all config maps
+	configMapsList := &corev1.ConfigMapList{}
+	err = nonCachingClient.List(context.TODO(), configMapsList, listOptions)
+	if err != nil {
+		logrus.Warn(getFailedToGetErrorMessageFor("Config Maps"))
+		noErrors = false
+	}
+	if configMapsList.Items != nil {
+		for _, cm := range configMapsList.Items {
+			labels := cm.GetLabels()
+			if labels != nil && labels[KubernetesInstanceLabelKey] != cheFlavor {
+				cm.ObjectMeta.Labels[KubernetesInstanceLabelKey] = cheFlavor
+				if err := nonCachingClient.Update(context.TODO(), &cm); err != nil {
+					logrus.Warn(getFailedToUpdateErrorMessage(cm.GetName(), reflect.TypeOf(cm).Name()))
+					noErrors = false
+				}
+			}
+		}
+	}
+
+	// Migrate all secrets
+	secretsList := &corev1.SecretList{}
+	err = nonCachingClient.List(context.TODO(), secretsList, listOptions)
+	if err != nil {
+		logrus.Warn(getFailedToGetErrorMessageFor("Secrets"))
+		noErrors = false
+	}
+	if secretsList.Items != nil {
+		for _, secret := range secretsList.Items {
+			labels := secret.GetLabels()
+			if labels != nil && labels[KubernetesInstanceLabelKey] != cheFlavor {
+				secret.ObjectMeta.Labels[KubernetesInstanceLabelKey] = cheFlavor
+				if err := nonCachingClient.Update(context.TODO(), &secret); err != nil {
+					logrus.Warn(getFailedToUpdateErrorMessage(secret.GetName(), reflect.TypeOf(secret).Name()))
+					noErrors = false
+				}
+			}
+		}
+	}
+
```

Comment:
We can refactor in more generic way.
```
  updateLables(&corev1.ConfigMapList{}, nonCachingClient)
  updateLables(&corev1.SecretList{}, nonCachingClient)

func updateLabels x(list client.ObjectList, nonCachingClient client.Client) {
 ...
}
```

Reply:
Looks like it is not possible to access `Items` of a generic list.